connect/runtime/src/main/java/org/apache/kafka/connect/util/ConnectUtils.java

```diff
@@ -65,4 +70,15 @@ static String lookupKafkaClusterId(Admin adminClient) {
                                        + "Check worker's broker connection and security properties.", e);
         }
     }
+
+    public static void addMetricsContextProperties(Map<String, Object> prop, WorkerConfig config, String clusterId) {
+        //add all properties start with "metrics.context"
+        for (String key : config.originals().keySet()) {
```

Comment:
you can use `config.originalWithPrefix(prefix, false)` to accomplish what you do here.